sycl/unittests/scheduler/AllocaLinking.cpp

```diff
@@ -0,0 +1,106 @@
+//==-------------- AllocaLinking.cpp --- Scheduler unit tests --------------==//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+#include "SchedulerTest.hpp"
+#include "SchedulerTestUtils.hpp"
+
+#include <helpers/PiMock.hpp>
+
+#include <iostream>
+
+using namespace sycl;
+
+static bool HostUnifiedMemory = false;
+
+static pi_result redefinedDeviceGetInfo(pi_device Device,
+                                        pi_device_info ParamName,
+                                        size_t ParamValueSize, void *ParamValue,
+                                        size_t *ParamValueSizeRet) {
+  if (ParamName == PI_DEVICE_INFO_HOST_UNIFIED_MEMORY) {
+    auto *Result = reinterpret_cast<pi_bool *>(ParamValue);
+    *Result = HostUnifiedMemory;
+  } else if (ParamName == PI_DEVICE_INFO_TYPE) {
+    auto *Result = reinterpret_cast<_pi_device_type *>(ParamValue);
+    *Result = PI_DEVICE_TYPE_CPU;
+  }
+  return PI_SUCCESS;
+}
+
+static pi_result
+redefinedMemBufferCreate(pi_context context, pi_mem_flags flags, size_t size,
+                         void *host_ptr, pi_mem *ret_mem,
+                         const pi_mem_properties *properties = nullptr) {
+  return PI_SUCCESS;
+}
+
+static pi_result redefinedMemRelease(pi_mem mem) { return PI_SUCCESS; }
+
+TEST_F(SchedulerTest, AllocaLinking) {
+  platform Plt{default_selector()};
+  if (Plt.is_host()) {
+    std::cout << "Not run due to host-only environment\n";
+    return;
```

Comment:
NIT: Probably it would be better to iterate over result of `sycl::platform::get_platforms`.

Reply:
Is the assumption that `default_selector` prefers non-host devices unsafe? If so, I think it would be better to apply this change to all unit tests separately.